ethereumj-core/src/main/java/org/ethereum/vm/hook/RootVmHook.java

```diff
@@ -39,35 +37,40 @@
 
     private static final Logger logger = LoggerFactory.getLogger("VM");
 
-    private final List<VMHook> hooks;
+    private VMHook[] hooks;
 
     @Autowired
-    public RootVmHook(Optional<List<VMHook>> hooks) {
-        this.hooks = hooks.orElseGet(ArrayList::new);
+    public RootVmHook(Optional<VMHook[]> hooks) {
```

Comment:
Not a blocker, but I would prefer using List instead of arrays as far as it's not critical. As soon as we added `hasHooks` flag this place is no more performance critical